homeassistant/components/advantage_air/climate.py

```diff
@@ -151,9 +156,14 @@ async def async_set_hvac_mode(self, hvac_mode):
 
     async def async_set_fan_mode(self, fan_mode):
         """Set the Fan Mode."""
-        await self.async_change(
-            {self.ac_key: {"info": {"fan": HASS_FAN_MODES.get(fan_mode)}}}
-        )
+        mode = HASS_FAN_MODES.get(fan_mode)
+        # MyAir5 has a unique automatic fan mode branded "myfan" that reports as "autoAA"
+        if (
+            self.coordinator.data["system"]["sysType"] == MYAIR5
+            and fan_mode == FAN_AUTO
+        ):
+            mode = MYAIR5_FAN_AUTO
+        await self.async_change({self.ac_key: {"info": {"fan": mode}}})
```

Comment:
Can this be handled in the upstream library instead?

Reply:
Yeah, I probably can do that instead if that's preferable.

Reply:
Absolutely.  The less Home Assistant knows about the implementation details, the better.  Anything we can move into the upstream library is a win as it means faster review and merges as well.